### PR TITLE
feat: support BigIntLiteral

### PR DESCRIPTION
**Pull request by ikatyang**
Ref: prettier/prettier#5545

- require `typescript@3.2.0-rc` (`bitInt` was added in TS 3.2)
- update to `@babel/parser@7.1.6` (`bigInt` was added in `7.0.0-beta.48`)

### CODE REVIEW

**Comment by JamesHenry** [comment]
This PR allowed me to catch the fact that a lot of ast-alignment tests are currently not being run: https://github.com/JamesHenry/typescript-estree/pull/36

**Comment by JamesHenry** [comment]
That PR has now been merged, it might be that is surfaces some issues with updating the babel parser version, sorry for the confusion.

I am hesitant to break away from our historically consistent approach of waiting until TypeScript versions become stable before supporting them.

Currently `semantic-release`, the library we use to handle releases here does not automatically support multiple release channels, but it looks like it will really soon! https://github.com/semantic-release/semantic-release/pull/991

Hopefully, therefore, for TypeScript 3.3 we will be able to automate everything, but for TypeScript 3.2, what I'd like to do is create a feature branch, and then once we are happy I will manually publish a release from it under a `next` tag. Prettier can then consume that until it become stable and we PR the feature branch into master.

Sound good?

**Comment by JamesHenry** [comment]
I've created the branch: https://github.com/JamesHenry/typescript-estree/tree/ts-3.2

**Comment by armano2** [comment]
@JamesHenry, why not going other way around? making default branch as stable, and following latests rc/beta changes on separate branch and releasing them as *.*.*-beta.X

**Comment by JamesHenry** [comment]
I think we are describing the same thing? I am suggesting the base of this PR be changed to the new branch I just created, which will be the source of a new, non-stable distribution on npm.

File: src/convert.ts [hunk 1]
```diff
@@ -2004,6 +2004,20 @@ export default function convert(config: ConvertConfig): ESTreeNode | null {
...
+      const raw = ast.text.slice(
+        (result as any).range[0],
+        (result as any).range[1]
+      );
+      const value = raw.slice(0, -1); // remove suffix `n`
+      Object.assign(result, {
+        type: AST_NODE_TYPES.BigIntLiteral,
+        raw,
+        value
+      });
```
Comment (ikatyang) [inline src/convert.ts]: I'm still not sure what should this node look like. `@babel/parser` parsed them as:

```ts
interface BigIntLiteral {
  type: "BigIntLiteral";
  value: string;
  extra: {
    raw: string;
    rawValue: string;
  }
}
```

but the alignment test rejected the `extra` field and expected:

```ts
interface BigIntLiteral {
  type: "BigIntLiteral";
  value: string;
  raw: string;
}
```

which is the current implementation.

Comment (JamesHenry) [reply src/convert.ts]: Hmm yeah seems there isn't quite a consensus yet: https://github.com/estree/estree/issues/169

There hasn't been any update on that issue for a while...

I think the least likely of all is having that extra object involved in ESTree, so maybe we should just ignore the AST comparison for now?

Comment (ikatyang) [reply src/convert.ts]: `extra` is already ignored, I just want to make sure if it's fine to output the node that's different from babel's. 😅

https://github.com/JamesHenry/typescript-estree/blob/8e7037599248b77d725ca53317231417f792f43e/tests/ast-alignment/utils.ts#L102-L105

Comment (JamesHenry) [reply src/convert.ts]: Yeah I think it's fine, I think we just have to accept that as experimental things get added they may be subject to change.

We can always publish new major versions in future if we need to amend it

**Pull request merged**